File: javascript/palindrome.js
const isPalindrome = (s) => {
    if (s.length < 2) {
      return true
    }
    if (s[0] !== s.slice(-1)) {
      return false
    } 
    return isPalindrome(s.slice(1, -1))
  }
  console.log(isPalindrome('abaca')) // false
  console.log(isPalindrome('aba')) // true

File: python/palindrome.py
def isPalindrome(s):
  if (len(s) < 2):
      return True
  if s[0] != s[-1:]:
      return False
   
  return isPalindrome(s[1:-1])


if __name__ == '__main__':
    print(isPalindrome('cabac'))
